src/index.ts

```diff
@@ -227,6 +228,11 @@ const config = {
     '@typescript-eslint/triple-slash-reference': ['error', { lib: 'never', path: 'never', types: 'never' }],
     '@typescript-eslint/type-annotation-spacing': 'error',
     '@typescript-eslint/unbound-method': ['error', { ignoreStatic: false }],
+    '@typescript-eslint/no-array-delete': 'error',
+    '@typescript-eslint/no-unsafe-unary-minus': 'error',
+    '@typescript-eslint/no-useless-template-literals': 'error',
+    '@typescript-eslint/prefer-find': 'error',
+    '@typescript-eslint/use-unknown-in-catch-callback-variable': 'error',
```

Comment:
If you could separate adding rules into a different PR, I would prefer that, please.

Reply:
I don't think I can separate upgrading `@typescript-eslint/parser` (and consecutively also `@typescript-eslint/eslint-plugin`) and introduced rules into separate PR's. These rules are a part of `@typescript-eslint/eslint-plugin` and tests started failing because of that upgrade so I had to add these. I am also not sure if I can introduce these rules prior upgrading (meaning I would add these rules one by one and then upgrade), and honestly it feels like more work for the same result. 🤷‍♂️

Reply:
Sorry. Perhaps the code is not self-explanatory enough.

In the test file there's a list of rules that are not yet considered. Rules that are new in the plugin should probably get added to that list when bumping the plugin.

Reply:
Gotcha. I have updated the code :v:

Reply:
`prefer-promise-reject-errors` is an "equivalent", however. Not sure how to skip it. The test explicitly tests for it to be there so I would say it should be there. Unless the test is wrong? Let me know or just approve the PR.

Reply:
@mightyiam is there anything else I should to do get this merged?